wikimedia/wikipedia-ios
# Native Editor - Add reference support

**Phabricator:** 
https://phabricator.wikimedia.org/T348072 (references portion)

### Notes
Implements syntax highlighting and actions for references.

### Test Steps
1. Open article from Staging scheme.
2. Confirm `<ref>` and `</ref>` tags are syntax highlighted green.
3. Place cursor within _inline_ `<ref></ref>` tags (that is, opening and closing doesn't span across multiple lines).
4. Confirm reference button shows selected state.
5. Tap reference button.
6. Confirm selection range expands to contained text and surrounding ref tags are removed. Confirm reference button shows unselected state.
7. Tap reference button again.
8. Confirm surrounding reference tags are added to selection.
